feature/613

AnimeOneLove2190 opened the pull request:
Add Page class
Add MovieSearchRequest class
Add MoviePageView class
Add GetPage controller in MovieController
Add GetPageAsync method in MovieService (not completed) 
Add GetPageAsync method in MovieDataAccess (not completed)

AnimeOneLove2190 committed:
feature/613
Add Page class
Add MovieSearchRequest class
Add MoviePageView class
Add GetPage controller in MovieController
Add GetPageAsync method in MovieService (not completed)
Add GetPageAsync method in MovieDataAccess (not completed)

artemlipinskiy commented on CinemaTicket.DataAccess/MovieDataAccess.cs:
```diff
@@ -40,14 +40,32 @@ public async Task<List<Movie>> GetMovieListAsync()
+                .OrderBy(x => x.CreatedOn)
+                .Skip(page * pageSize)
+                .Take(pageSize)
+                .Where(x => movieName.Contains(x.Name))
```
1. Название в бд должно содержать поисковый запрос, а не наоборот
2. Привести поисковый запрос к нижнему регистру и название в бд

.Where(x => x.Name.ToLower().Contains(movieName.ToLower()))

AnimeOneLove2190 committed:
feature/613/2
GetPageAsync in MovieService method changed
GetPageAsync in MovieDataAccess method changed

artemlipinskiy commented on CinemaTicket.DataAccess/MovieDataAccess.cs:
```diff
@@ -40,14 +41,51 @@ public async Task<List<Movie>> GetMovieListAsync()
+            }
+            if (movieSearchRequest.GenreNames.Count > 0)
+            {
+                query = query.Where(x => x.Genres.Any(g => movieSearchRequest.GenreNames.Contains(g.Name)));
```
Передаваться должны id жанров, не названия. Если жанры указаны, то они должны быть все у фильма

artemlipinskiy commented on CinemaTicket.DataAccess/MovieDataAccess.cs:
```diff
@@ -40,14 +41,51 @@ public async Task<List<Movie>> GetMovieListAsync()
+            var query = cinemaManagerContext.Movies.Include(x => x.Genres).AsQueryable();
+            if (!string.IsNullOrEmpty(movieSearchRequest.MovieName) || !string.IsNullOrWhiteSpace(movieSearchRequest.MovieName))
+            {
+                query = query.Where(x => x.Name.ToLower().Contains(movieSearchRequest.MovieName.ToLower()) || movieSearchRequest.MovieName.ToLower().Contains(x.Name.ToLower()));
```
x.Name.ToLower().Contains(movieSearchRequest.MovieName.ToLower()) достаточно

artemlipinskiy commented on CinemaTicket.DataAccess/MovieDataAccess.cs:
```diff
@@ -40,14 +41,51 @@ public async Task<List<Movie>> GetMovieListAsync()
+            }
+            if (!string.IsNullOrEmpty(movieSearchRequest.Description) || !string.IsNullOrWhiteSpace(movieSearchRequest.Description))
+            {
+                query = query.Where(x => x.Name.ToLower().Contains(movieSearchRequest.Description.ToLower()) || movieSearchRequest.Description.ToLower().Contains(x.Name.ToLower()));
```
x.Description.ToLower(), не Name

AnimeOneLove2190 committed:
feature/613/3
Changes applied

AnimeOneLove2190 committed:
feature/613/4
Changes applied
